File: src/main/java/com/kurtraschke/gtfsrtdump/output/CsvOutput.java
package com.kurtraschke.gtfsrtdump.output;

import com.google.transit.realtime.GtfsRealtime;
import com.kurtraschke.gtfsrtdump.output.csv.FeedParts;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;

@Command(name = "csv", description = "CSV output")
public class CsvOutput extends OutputMethod {

    @Parameters(index = "0", description = "Feed part to extract to CSV. Valid values: ${COMPLETION-CANDIDATES}")
    FeedParts selectedFeedPart;

    @Override
    public Integer format(GtfsRealtime.FeedMessage fm) {
        selectedFeedPart.generateOutput(fm);
        return 0;
    }
}

File: src/main/kotlin/com/kurtraschke/gtfsrtdump/output/csv/CsvOutput.kt
package com.kurtraschke.gtfsrtdump.output.csv

import com.google.transit.realtime.GtfsRealtime.FeedMessage
import com.kurtraschke.gtfsrtdump.output.OutputMethod
import picocli.CommandLine.Command
import picocli.CommandLine.Parameters
import java.io.PrintWriter

@Command(
    name = "csv", description = ["CSV output"]
)
class CsvOutput : OutputMethod() {

    @Parameters(
        index = "0", description = ["Feed part to extract to CSV. Valid values: \${COMPLETION-CANDIDATES}"]
    )
    private lateinit var selectedFeedPart: FeedParts

    override fun format(fm: FeedMessage, w: PrintWriter): Int {
        selectedFeedPart.generateOutput(fm, w)
        return 0
    }
}
